os/arch/xtensa/src/esp32/esp32_gpio_lowerhalf.c

```diff
@@ -144,8 +144,10 @@ static int esp32_gpio_setdir(FAR struct gpio_lowerhalf_s *lower, unsigned long a
 	priv->pincfg &= ~FUNCTION_MASK;
 
 	if (arg == GPIO_DIRECTION_OUT) {
+		priv->pincfg &= ~INPUT;
```

Comment:
Is esp32 can't use input if direction is input?
As I know when I/O Pin set as an output, it is possible to read as an input pin, I mean some of chips supports it.

Reply:
If It supports it, only below code can be accepted, I mean.

Reply:
Confused, ignore my comments.